Updated airgap.md with steps to setup rancher and harvester in air-gapped env (#371)
harvester/docs: docs/airgap.md

The Harvester ISO image contains all the packages to make it work in an air gapped environment.

## Working without HTTP Proxy

In this example, we use KVM to provision Harvester and integrate it with Rancher.

> ### Setup offline Harvester

Reviewer: ```suggestion
### Set up offline Harvester
```

    sudo cp certs/domain.crt /etc/docker/certs.d/myregistry.local:5000/domain.crt
    ```

1. Start the private registry.

    ```
    docker run -d \
    -p 5000:5000 \
    --restart=always \
    --name registry \
    -v "$(pwd)"/certs:/certs \
    -v "$(pwd)"/registry:/var/lib/registry \
    -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
    -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key \
    registry:2
    ```

1. Add `myregistry.local` to `/etc/hosts`. Remember to change `192.168.0.50` to your private IP.

    ```
    # vim /etc/hosts
    192.168.0.50 myregistry.local
    ```

1. Create a `get-rancher` script.

    ```
    # vim get-rancher
    #!/bin/bash
    if [[ $# -eq 0 ]] ; then
        echo 'This requires you to pass a version for the url like "v2.6.4"'
        exit 1
    fi
    wget https://github.com/rancher/rancher/releases/download/$1/rancher-images.txt
    wget https://github.com/rancher/rancher/releases/download/$1/rancher-load-images.sh
    wget https://github.com/rancher/rancher/releases/download/$1/rancher-save-images.sh
    chmod +x ./rancher-save-images.sh
    chmod +x ./rancher-load-images.sh
    ```

> 1. Make `get-rancher` script be excutable.

Reviewer: ```suggestion
1. Make the `get-rancher` script excutable.
```

    ```
    chmod +x get-rancher
    ```

1. Download `rancher-images.txt`, `rancher-load-images.sh` and `rancher-save-images.sh`.

    ```
    ./get-rancher v2.6.4
    ```

1. Add cert-manager images to `rancher-images.txt`.

    ```
    helm repo add jetstack https://charts.jetstack.io/
    helm repo update
    helm fetch jetstack/cert-manager --version v1.7.1
    helm template ./cert-manager-v1.7.1.tgz | awk '$1 ~ /image:/ {print $2}' | sed s/\"//g >> ./rancher-images.txt
    ```

1. Sort `rancher-images.txt`.

    ```
    sort -u rancher-images.txt -o rancher-images.txt
    ```

1. Get images. This step may take 1 to 2 hours depending on your network speed.

    ```
    ./rancher-save-images.sh --image-list ./rancher-images.txt
    ```

> 1. Load images to local registry.

Reviewer: ```suggestion
1. Load images to the local registry.
```

    ```
    ./rancher-load-images.sh --image-list ./rancher-images.txt --registry myregistry.local:5000
    ```

1. Create a `get-k3s` script.

    ```
    # vim get-k3s
    #!/bin/bash
    if [[ $# -eq 0 ]] ; then
        echo 'This requires you to pass a version for the url like "v1.23.4+k3s1"'
        exit 1
    fi
    wget https://github.com/k3s-io/k3s/releases/download/$1/k3s-airgap-images-amd64.tar
    wget https://github.com/k3s-io/k3s/releases/download/$1/k3s
    wget https://get.k3s.io/ -O install.sh
    chmod +x ./k3s
    chmod +x ./install.sh
    ```

1. Make `get-k3s` excuteable.

    ```
    chmod +x get-k3s
    ```

1. Download `k3s-airgap-images-amd64.tar`, `k3s` and `install.sh`.

    ```
    ./get-k3s v1.23.4+k3s1
    ```

1. Download Rancher.

    ```
        helm repo add rancher-latest https://releases.rancher.com/server-charts/latest
        helm fetch rancher-latest/rancher --version=v2.6.4
    ```

1. Download `cert-manager-crds.yaml`.

    ```
    mkdir cert-manager
    curl -L -o cert-manager/cert-manager-crd.yaml https://github.com/jetstack/cert-manager/releases/download/v1.7.1/cert-manager.crds.yaml
    ```

> 1. Remove the `vagrant-libvirt` network interface. Once the required resources download, you can remove the network.

Reviewer: ```suggestion
1. Once the required resources are downloaded, you can remove the `vagrant-libvirt` network interface. 
```

    ```

1. Create a `cattle-system` namespace.

    ```
    kubectl create ns cattle-system
    ```

1. Create a `tls.sa` secret.

    ```
    kubectl -n cattle-system create secret generic tls-ca \
    --from-file=cacerts.pem=./cacerts.pem
    ```

1. Create a `tls-rancher-ingress` secret.

    ```
    kubectl -n cattle-system create secret tls tls-rancher-ingress \
    --cert=tls.crt \
    --key=tls.key
    ```

1. Generate Rancher YAML files.

    ```
    helm template rancher ./rancher-2.6.4.tgz --output-dir . \
        --no-hooks \
        --namespace cattle-system \
        --set hostname=myrancher.local \
        --set rancherImageTag=v2.6.4 \
        --set rancherImage=myregistry.local:5000/rancher/rancher \
        --set systemDefaultRegistry=myregistry.local:5000 \
        --set useBundledSystemChart=true \
        --set ingress.tls.source=secret \
        --set privateCA=true
    ```

1. Install Rancher.

    ```
    kubectl -n cattle-system apply -R -f ./rancher
    ```

1. Add `myrancher.local` to `/etc/hosts`. Remember to change `192.168.0.50` to your private IP.

    ```
    # vim /etc/hosts
    192.168.0.50 myregistry.local myrancher.local
    ```

> ### Integrate Harvester with Rancher in air-gapped environment

Reviewer: ```suggestion
### Integrate Harvester with Rancher in an air-gapped environment
```
